l10n_es_aeat_verifactu/models/account_move.py

```diff
@@ -65,3 +103,52 @@ def _get_document_serial_number(self):
         if self.thirdparty_invoice:
             serial_number = self.thirdparty_number[0:60]
         return serial_number
+
+    def _get_verifactu_issuer(self):
+        return self.company_id.partner_id._parse_aeat_vat_info()[2]
+
+    def _get_verifactu_document_type(self):
+        return self.verifactu_document_type or "F1"
+
+    def _get_verifactu_amount_tax(self):
+        return self.amount_tax
+
+    def _get_verifactu_amount_total(self):
+        return self.amount_total
+
+    def _get_verifactu_previous_hash(self):
+        # TODO store it? search it by some kind of sequence?
```

Comment:
Indeed Odoo chose to create a sequence when creating the journal and uses the sequence to find the previous entry.

https://github.com/odoo/odoo/blob/d8a20717c830a7cddaad4889bec182f0fa5f786a/addons/account/models/account_move.py#L2548-L2551

https://github.com/odoo/odoo/blob/d8a20717c830a7cddaad4889bec182f0fa5f786a/addons/account/models/account_journal.py#L950

Reply:
Yes, but the problem in this case is that the previous record is not the previous one from the same journal. A possible solution is to implement a unique sequence for all invoices, but we also need to know if they have been sent and received by the AEAT.

Reply:
Lo que se me ocurre es una especie de cola, así se elimina el problema de la concurrencia (y se puede tener tantas secuencias como diarios). La cola FIFO guarda la referencia o hash de la última factura en un campo global por orden de inserción.

Faltaría decidir cuando meter la factura en la cola que generará el hash y la enviará a AEAT. No sé si por la ley hay obligatoriedad de hacerlo cuando se le da a enviar factura o solamente al guardar, o pasa como el SII que puede ser manual. En cualquier caso sería meterlo en la cola desde cualquier diario y encadenar facturas de diferentes diarios, validadas por diferentes usuarios, o demás **concurrencias**, gracias a la cola no supondrían un problema.

es por dar ideas.

Reply:
Me resulta muy extraño que esa factura anterior tenga que ser global en lugar de por diario. La AEAT es muy consciente de las cuestiones técnicas asociadas a tener más de una secuencia de numeración (como múltiples canales de venta).

Reply:
Según mi interpretación de la documentación, el encadenamiento es por _SIF_ (Sistema Informático de Facturación).

![imagen](https://github.com/user-attachments/assets/5b1cb2ad-8588-4cd9-b796-97ecbdfaae34)

```
Caso 2: registro de facturación –en este caso de alta–
con registro de facturación anterior existente en el SIF
(siendo, por tanto, el segundo registro o sucesivo)
```

Lo interpreto de ese modo porque dice `factura anterior existente en el SIF`.

En caso de ser mi interpretación correcta, entiendo que habría que crear un SIF para la facturación de Odoo y uno por TPV, si no recuerdo mal así es como funciona en TicketBAI.

Reply:
Si por eso interpretamos que el encadenamiento era con el anterior enviado y eso excluye el orden por diario pero realmente tampoco habla de series en ningun lado que yo haya visto...